foxpuppet/expected.py

```diff
@@ -0,0 +1,22 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+class new_browser_window_is_opened(object):
+    """
+    An expectation for checking that a new window is found after requesting
+    it to be opened
+
+    :returns: A BrowserWindow Object
+    """
+
+    def __init__(self, selenium, handles):
+        self.selenium = selenium
+        self.handles = handles
+
+    def __call__(self, *args, **kwargs):
+        handles = list(set(self.selenium.window_handles) - set(self.handles))
+        if len(handles) == 1:
+            from .windows.browser import BrowserWindow
+            return BrowserWindow(self.selenium, handles[0])
```

Comment:
Agreed. I'd see us deprecating this in the future, which is why I was keen to have the window type mentioned in the name. We'd ideally have a generic expectation that returned the specific window type.